subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/junitplatform/JUnitPlatformIntegrationTest.groovy

```diff
@@ -409,4 +409,48 @@ public class StaticInnerTest {
         "w/o filters"  | []
         "with filters" | ['--tests', 'JUnitJupiterTest']
     }
+
+    def "creates LauncherSession before loading test classes"() {
```

Comment:
This should probably live in `JUnitPlatformLauncherSessionListenerIntegrationTest`

Reply:
Done in 0934c7ae0f878195f665db3dbd1ef64beead3bf1.